lib/modules/manager/circleci/extract.ts

```diff
@@ -1,80 +1,50 @@
 import { logger } from '../../../logger';
-import { newlineRegex, regEx } from '../../../util/regex';
+import { coerceArray } from '../../../util/array';
+import { parseSingleYaml } from '../../../util/yaml';
 import { OrbDatasource } from '../../datasource/orb';
 import * as npmVersioning from '../../versioning/npm';
 import { getDep } from '../dockerfile/extract';
 import type { PackageDependency, PackageFileContent } from '../types';
+import { CircleCiFile } from './schema';
 
 export function extractPackageFile(
   content: string,
   packageFile?: string,
 ): PackageFileContent | null {
   const deps: PackageDependency[] = [];
   try {
-    const lines = content.split(newlineRegex);
-    for (let lineNumber = 0; lineNumber < lines.length; lineNumber += 1) {
-      const line = lines[lineNumber];
-      const orbs = regEx(/^\s*orbs:\s*$/).exec(line);
-      if (orbs) {
-        logger.trace(`Matched orbs on line ${lineNumber}`);
-        let foundOrbOrNoop: boolean;
-        do {
-          foundOrbOrNoop = false;
-          const orbLine = lines[lineNumber + 1];
-          logger.trace(`orbLine: "${orbLine}"`);
-          const yamlNoop = regEx(/^\s*(#|$)/).exec(orbLine);
-          if (yamlNoop) {
-            logger.debug('orbNoop');
-            foundOrbOrNoop = true;
-            lineNumber += 1;
-            continue;
-          }
-          const orbMatch = regEx(/^\s+([^:]+):\s(.+?)(?:\s*#.*)?$/).exec(
-            orbLine,
-          );
-          if (orbMatch) {
-            logger.trace('orbMatch');
-            foundOrbOrNoop = true;
-            lineNumber += 1;
-            const depName = orbMatch[1];
-            const [orbName, currentValue] = orbMatch[2].split('@');
-            const dep: PackageDependency = {
-              depType: 'orb',
-              depName,
-              currentValue,
-              datasource: OrbDatasource.id,
-              packageName: orbName,
-              commitMessageTopic: '{{{depName}}} orb',
-              versioning: npmVersioning.id,
-            };
-            deps.push(dep);
-          }
-        } while (foundOrbOrNoop);
-      }
-      const match = regEx(/^\s*-? image:\s*'?"?([^\s'"]+)'?"?\s*$/).exec(line);
-      if (match) {
-        const currentFrom = match[1];
-        const dep = getDep(currentFrom);
-        logger.debug(
-          {
-            depName: dep.depName,
-            currentValue: dep.currentValue,
-            currentDigest: dep.currentDigest,
-          },
-          'CircleCI docker image',
-        );
-        dep.depType = 'docker';
-        dep.versioning = 'docker';
-        if (
-          !dep.depName?.startsWith('ubuntu-') &&
-          !dep.depName?.startsWith('windows-server-') &&
-          !dep.depName?.startsWith('android-') &&
-          dep.depName !== 'android'
```

Comment:
you dropped this condition, it should stay because those values are not resolvable docker images

Reply:
This on purpose this is not needed anymore. 

Before that it has been used to cover `jobs.release_docker.machine.image` as the regexes looked for `image:`

https://github.com/renovatebot/renovate/blob/34c56e3fe1c4048cfadab6a706553d488c1ef0a0/lib/modules/manager/circleci/__fixtures__/config3.yml#L15-L17